lib/policyEvaluator/principal.js

```diff
@@ -35,7 +35,8 @@ class Principal {
                 // In case of anonymous NotPrincipal, this will neutral everyone
                 return 'Neutral';
             }
-            if (!Principal._evaluateCondition(params, statement)) {
+            const conditionEval = Principal._evaluateCondition(params, statement);
+            if (!conditionEval || conditionEval.allow === false) {
```

Comment:
Could write `!conditionEval.allow` (it would also catch any result that is not explicitly allowed)`

Reply:
I had that originally and the check was failing. Not sure why, but this seems to work

Reply:
`conditionEval.allow === false` implies `!conditionEval.allow` (but not the other way) so it should make this test broader, but it should not fail with `!conditionEval.allow` where it succeeds with `conditionEval.allow === false`. Can you check this to understand exactly what is happening? I think it's important since it looks like a critical piece of code to me.